### PR TITLE
T13672

### PR DESCRIPTION
**Pull request by wgreenberg**
Implements `kermit stat` and `kermit crosslink` utility functions

https://phabricator.endlessm.com/T13672

### CODE REVIEW

File: js/tools/kermit.js [hunk 1]
```diff
@@ -76,6 +79,26 @@ function grep (path, pattern) {
...
+    let shard = get_shard_for_path(path);
+    let table_record = shard.find_record_by_hex_name(LINK_TABLE_ID);
+    let dictionary;
+    if (table_record) {
+        dictionary = table_record.data.load_as_dictionary();
+    } else {
+        fail_with_message('No dictionary record found in this shard!');
+    }
+    let ekn_id = dictionary.lookup_key(url);
+    if (!ekn_id) {
```
Comment (magcius) [inline js/tools/kermit.js]: In if/else statements, I prefer not inverting the conditional. Makes it easier to read, IMO

File: js/tools/kermit.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ const USAGE = [
     '       kermit dump <shard path> <ekn id> <data|metadata|another blob name>',
     '       kermit query <app_id> "<querystring>"',
     '       kermit stat <shard path>',
+    '       kermit lookup <shard path> <url>',
```
Comment (magcius) [inline js/tools/kermit.js]: I'm not 100% sure I like the name "lookup". Perhaps `kermit crosslink`?

File: js/tools/kermit.js [hunk 3]
```diff
@@ -76,6 +79,26 @@ function grep (path, pattern) {
     });
 }
 
+// This hash is derived from sha1('link-table'), and for now is the hardcoded
+// location of link tables for all shards.
+const LINK_TABLE_ID = '4dba9091495e8f277893e0d400e9e092f9f6f551';
```
Comment (magcius) [inline js/tools/kermit.js]: I thought we did the `GLib.Checksum.new` inline, but I just checked domain.js, and I guess not. OK.

**Comment by wgreenberg** [comment]
@magcius addressed those comments

File: js/tools/kermit.js [hunk 4]
```diff
@@ -73,6 +79,255 @@ function grep (path, pattern) {
...
+function crosslink (path, url) {
+    let shard = get_shard_for_path(path);
+    let table_record = shard.find_record_by_hex_name(LINK_TABLE_ID);
+    let dictionary;
+    if (table_record) {
+        dictionary = table_record.data.load_as_dictionary();
+    } else {
+        fail_with_message('No dictionary record found in this shard!');
```
Comment (magcius) [inline js/tools/kermit.js]: "No link table"

**Pull request merged**